src/onnx/parse_slice.cpp

```diff
@@ -144,13 +144,18 @@ struct parse_slice : op_parser<parse_slice>
             sd.op.axes = axes;
         }
 
-        if(not sd.steps.empty())
+        if(std::any_of(sd.steps.begin(), sd.steps.end(), [](auto s) { return s != 1; }))
         {
             if(sd.op.starts.empty() or sd.op.ends.empty())
-                MIGRAPHX_THROW("PARSE_SLICE: steps and variable starts and ends is not supported");
+                MIGRAPHX_THROW(
+                    "PARSE_SLICE: steps and variable starts and/or ends is not supported");
             if(sd.op.axes.empty())
                 MIGRAPHX_THROW("PARSE_SLICE: steps and variable axes is not supported");
         }
+        else
+        {
+            sd.steps.clear();
```

Comment:
why steps are being cleared ? Is it because it would have all its values as 1 ?

Reply:
Mainly to get around doing the next new lines for a vector of all 1's.

Reply:
Why do even need  to clear it ? It if falls into `else` clause it would have all values 1.